### ISSUE TITLE
CRC Error unzipping inms.csv on Windows

### ISSUE THREAD

**Issue by rprouse** [red-4/curious-moon#23]
I downloaded the book with data. Attempting to unzip with 7-Zip on Windows 10 resulted in a CRC error in `inms.csv`. I then tried downloading the data from the link in the book, but I get the same error.

![image](https://user-images.githubusercontent.com/493828/35930833-f4735204-0c00-11e8-8b6b-edd6d1659298.png)

Next, I tried unzipping using the built in Windows Explorer zip, that gives an `Unspecified Error`

![image](https://user-images.githubusercontent.com/493828/35931062-9fcb4e90-0c01-11e8-9494-3c4e16925d9f.png)

The book being all about the command line, I dropped to Git Bash. Using `unzip` worked. Command line saves the day 😄 

It seems to be a Windows thing though, I also tried the 7-Zip command line, but it failed,

```bat
λ  & 'C:\Program Files\7-Zip\7z.exe' x .\cassini_data.zip

7-Zip [64] 16.04 : Copyright (c) 1999-2016 Igor Pavlov : 2016-10-04

Scanning the drive for archives:
1 file, 640336154 bytes (611 MiB)

Extracting archive: .\cassini_data.zip
--
Path = .\cassini_data.zip
Type = zip
Physical Size = 640336154

ERROR: CRC Failed : data\INMS\inms.csv

Sub items Errors: 1

Archives with Errors: 1

Sub items Errors: 1
```

**Comment by rprouse** [comment]
To check that it isn't something specific about the file, I created a new ZIP using 7-Zip and it uncompressed fine.

**Comment by danners** [comment]
I can reproduce with winzip + 7zip, the csv when extracted is incomplete.

**Comment by ttrunck** [comment]
I can also repro with the build-in Windows Explorer unzip.
But a recent version of 7-zip (18.01 (x64)) was able to unzip it fine.

Do you have a hash of that file so we can check it's correct?
I have that:
```
$ sha1sum inms.csv
c814bf5570742ec352994e3b32376ee0fa694b50  inms.csv
```

**Comment by robconery** [comment]
This is a bug in Windows/Windows explorer that I'm trying to track down. I do know that if you use the command line client for 7zip it works fine; also Linux unzips OK too. Having a hard time figuring this one out.

**Comment by rprouse** [comment]
Odd, the 7-Zip command line didn't work for me. See the output at the end of my initial report. Older version?

**Comment by robconery** [comment]
Hmmm. Damn. This one sucks. I'm doing another push in the coming days so maybe I'll 7-zip the data on Windows and that will fix.

**Comment by rprouse** [comment]
I just tested 7-Zip 18.01 and it works fine from the command line and using the UI, so it must have been an issue with the older version. I still get the error with the Windows built-in Zip for both the book and data and just the straight data files.

To clarify my 2nd comment from Feb 7, I re-compressed the data using 7-Zip and it re-extracted fine, so your plan of compressing the new upload with 7-Zip on windows should resolve this issue.

**Comment by robconery** [comment]
I'm uploading a 7-zipped archive right now which will hopefully solve this. Thanks for the help!

**Comment by TechnoTone** [comment]
Hey @robconery, I just ran into this same issue (tried on 2 laptops both running Windows 10). I think I've gotten around it by using 7-zip instead but it's very strange. Just thought you might like to know.

**Comment by robconery** [comment]
Thanks - but to be sure I understand - what are you wanting me to know? So far the only thing I can come up with is Windows doesn't like large files unless you use 7-zip.

**Comment by TechnoTone** [comment]
So, two things...

1st, in your post above where you said you have uploaded a 7-zipped archive instead, I read this (along with the `fixed` label) to mean that the issue was resolved.

2nd, I have the files that 7-zip extracted successfully. As a test, I have used Windows to re-compress them into a new zip file of about 620MB. I then re-extracted these files to a new location and Windows was absolutely fine with it.

Perhaps there's something strange with the original file's zip format that Windows doesn't like but it's certainly bizarre.

Happy to upload the file if it is of any use to you.
